vars/abapEnvironmentPipelineStageIntegrationTests.groovy

```diff
@@ -17,8 +21,23 @@ void call(Map parameters = [:]) {
     def script = checkScript(this, parameters) ?: this
     def stageName = parameters.stageName?:env.STAGE_NAME
 
+    Map config = ConfigurationHelper.newInstance(this)
+         .loadStepDefaults()
+         .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+         .mixinStageConfig(script.commonPipelineEnvironment, stageName, STEP_CONFIG_KEYS)
+         .mixin(parameters, PARAMETER_KEYS)
+         .use()
+
     piperStageWrapper (script: script, stageName: stageName, stashContent: [], stageLocking: false) {
-        cloudFoundryCreateService script: parameters.script
+        try {
+            cloudFoundryCreateService script: parameters.script
+        } catch (Exception e) {
+            script.currentBuild.result = 'UNSTABLE'
+        }
+        if (config.confirmDeletion) {
+            input message: "Pipeline status is not successful. Once you proceed, the system will be deleted."
+        }
+        cloudFoundryDeleteService script: parameters.script
```

Comment:
Regarding the discussion about the default value of `configDeletion`: IMO leaving it to false also makes up for way better readability here

👍

Reply:
I would even go so far that for the *Integration Tests* stage we should not provide the possibility to delete the system without confirmation.

The confirmation message should also be different: "e.g. Add-On Product was installed successfully?" Because that is actually the most essential part here, to confirm that the add-on product was installed successfully. 

The partner might execute more "integration tests" here, but that should be optional. Most important is here: Only if the add-on product is installable, a productive target vector should be published in the *Publish* stage. And the question for this add-on installation test should be asked here.